### PR TITLE
Implement ChainHealthiness query

### PR DESCRIPTION
**Pull request by hochiw**
## Features
- Implements a query on graphql server for the chain healthiness based on the current time, average block time and the current block timestamp

refs #46

### CODE REVIEW

File: graphql-server/pkg/resolvers/chain.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	timeSinceLastBlock := now.Sub(currentBlock.Timestamp).Seconds()
+	chainStatus := models.ChainStatusOnline
+	// See chain as halted if missed 20+ blocks
+	if timeSinceLastBlock > 20*averageBlockTime.AverageTime {
+		chainStatus = models.ChainStatusHalted
+		// See chain as congested if missed 5 blocks
```
Comment (chihimng) [inline graphql-server/pkg/resolvers/chain.go:30]: bad indentation?

**Review by chihimng — APPROVED** [review]

File: graphql-server/pkg/resolvers/chain.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	currentBlock, err := pkgContext.GetQueriesFromCtx(ctx).Block.QueryLatestBlock()
+	if err != nil {
+		return nil, err
+	}
+	averageBlockTime, err := pkgContext.GetQueriesFromCtx(ctx).Chain.QueryAvergeBlockTime()
+	if err != nil {
+		return nil, err
+	}
+
+	timeSinceLastBlock := now.Sub(currentBlock.Timestamp).Seconds()
```
Comment (chihimng) [inline graphql-server/pkg/resolvers/chain.go:25]: i have some doubts on latency for bdjuno to index the block....

Comment (hochiw) [reply graphql-server/pkg/resolvers/chain.go:25]: timeSinceLastBlock is always 10+ secs actually

Comment (chihimng) [reply graphql-server/pkg/resolvers/chain.go:25]: how about using time between block n-1 and n-2 for congestion / healthy; & if only show halted if the last block time is way off from wall time?

**Comment by hochiw** [comment]
@chihimng updated :adore2x:

**Pull request merged**